plugins/error-augmentor/plugin.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	goplugin "github.com/hashicorp/go-plugin"
+	"gopkg.in/yaml.v2"
+
+	buildeventstream "aspect.build/cli/bazel/buildeventstream/proto"
+	"aspect.build/cli/pkg/ioutils"
+	"aspect.build/cli/pkg/plugin/sdk/v1alpha2/config"
+)
+
+func main() {
+	goplugin.Serve(config.NewConfigFor(NewDefaultPlugin()))
+}
+
+type ErrorAugmentorPlugin struct {
+	properties             []byte
+	hintMap                map[*regexp.Regexp]string
+	yamlUnmarshalStrict    func(in []byte, out interface{}) (err error)
+	helpfulHints           *helpfulHintSet
+	helpfulHintsMutex      sync.Mutex
+	errorMessages          chan string
+	errorMessagesWaitGroup sync.WaitGroup
+}
+
+func NewDefaultPlugin() *ErrorAugmentorPlugin {
+	return NewPlugin()
+}
+
+func NewPlugin() *ErrorAugmentorPlugin {
+	return &ErrorAugmentorPlugin{
+		properties:          nil,
+		hintMap:             map[*regexp.Regexp]string{},
+		yamlUnmarshalStrict: yaml.UnmarshalStrict,
+		helpfulHints:        &helpfulHintSet{nodes: make(map[helpfulHintNode]struct{})},
+	}
+}
+
+// will be used to unmarshal plugin properties specific to this plugin
+type errorMappings struct {
+	ErrorMappings map[string]string `yaml:"error_mappings"`
+}
+
+func (plugin *ErrorAugmentorPlugin) SetupHook(
+	properties []byte,
+) error {
+	plugin.properties = properties
+
+	var processedProperties errorMappings
+	if err := plugin.yamlUnmarshalStrict(properties, &processedProperties); err != nil {
+		return fmt.Errorf("failed to setup: failed to parse properties: %w", err)
```

Comment:
+1 - perfect error wrapping message with the first part being the current context, the second part an additional context from the generic unmarshal function, and the actual error from the call.

Reply:
:)